tests/test_wait.py

```diff
@@ -0,0 +1,73 @@
+from datetime import timedelta
```

Comment:
It would be so amazing to have the test written as not sleeping actually but somehow simulating time. I don't have any constructive point on this, as queueing is implemented in python we can't really do anything here, but if you have any idea, please share it. :)

Reply:
What's wrong with sleeping for 1 second? All tests pass in several seconds which is fine. And yes, I don't have ideas.

Reply:
According to my experience, having anything related to time during the tests add flakiness as it is not explicit (for example the reason why you cannot compare the sleeping time to == 1 in the test code).
Also, waiting for an event during the test may look ok when it runs in 6 seconds but when there are a lot of tests and you need to optimize, you can't solve it by putting more hardware for it or tuning the OS because the sleep will take that much time. This is wasted time I think.
In normal cases when you are sleeping for a given time (which is not the case currently) you can mock the sleeping or factor it out to a function but this is not the case.
I have other tests in this code which sleep for at most 1 seconds and I couldn't improve them because sleeping is wired into other libraries (stdlib or werkzeug, I don't remember exactly).

Reply:
OK, I get the point, but I don't know how to make a workaround.